CMakeLists.txt

```diff
@@ -9,25 +9,25 @@ if (NOT BLT_CXX_STD)
     set(BLT_CXX_STD "c++14" CACHE STRING "")
 endif()
 
-include(cmake/blt_boilerplate.cmake)
-
+set(DESUL_DEPENDENT_OPTIONS ENABLE_OPENMP ENABLE_CUDA ENABLE_HIP ENABLE_SYCL ENABLE_TESTS)
+foreach (option ${DESUL_DEPENDENT_OPTIONS})
+   if (${DESUL_${option}})
+      if (NOT ${option})
+         message(FATAL_ERROR "DESUL_${option} set to On, but ${option} is Off. Please set ${option} to On enable this feature.")
+      endif ()
+   endif ()
+endforeach ()
+
+
+include(CMakeDependentOption)
+option(ENABLE_SYCL "Enable SYCL support" OFF) # option does not actually yet exist on the BLT side
+cmake_dependent_option(DESUL_ENABLE_CUDA "Build with CUDA support" On "ENABLE_CUDA" Off)
```

Comment:
I am not a big fan of mixing the spelling `ON/OFF` (line above) and "On/Off"
I would prefer if we stuck to uppercase.

Reply:
Sure, no problem.